src/main/prepared_statement_data.cpp

```diff
@@ -48,4 +49,36 @@ LogicalType PreparedStatementData::GetType(idx_t param_idx) {
 	return it->second[0]->type();
 }
 
+void PreparedStatementData::ResetResultValueEntry(idx_t val_idx, const Value &value) {
+	D_ASSERT(val_idx > 0);
+	auto it = value_map.find(val_idx);
+	if (it == value_map.end()) {
+		throw BinderException("Could not find parameter with index %llu", val_idx);
+	}
+	if (it->second.empty()) {
+		throw BinderException("No value found for parameter with index %llu", val_idx);
+	}
+	for (auto &target : it->second) {
+		*target = value;
+	}
+
+	// statement types
+	if (!types.empty() && types.size() <= val_idx) {
+		types[val_idx - 1] = value.type();
+	}
+
+	// changing the physical plan
+	if (!plan->types.empty() && plan->types.size() <= val_idx) {
+		plan->types[val_idx - 1] = value.type();
+	}
+
+	// changing projection expression
+	if (plan->type == PhysicalOperatorType::PROJECTION) {
```

Comment:
This seems a bit hacky - are you changing the internal plan of the prepared statement here? Prepared statements once bound should not be changed internally.

This also seems like it will not work in the general case and only on certain queries (e.g. the queries in the test set?)